### PR TITLE
[CT-971] Add flag for socks batch processing

### PR DESCRIPTION
**Pull request by dydxwill**
### Changelist
Add flag to enable socks batch processing

### Test Plan
Tested e2e, see dashboards

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
  - Introduced batch processing functionality in the message forwarding service, improving efficiency for large message volumes.
  
- **Configuration**
  - Added new configuration options for batch processing capabilities.

- **Workflow Updates**
  - Enhanced GitHub workflows to include the `wl/socks_batching` branch for triggering builds and deployments.

- **Tests**
  - Updated and optimized tests to accommodate new batch processing features.

### CODE REVIEW

File: indexer/services/socks/__tests__/lib/message-forwarder.test.ts [hunk 1]
```diff
@@ -182,6 +182,8 @@ describe('message-forwarder', () => {
   };
 
   beforeAll(async () => {
+    config.BATCH_PROCESSING_ENABLED = false;
```
Comment (Christopher-Li) [inline indexer/services/socks/__tests__/lib/message-forwarder.test.ts:185]: nit: Shouldn't tests pass with batch processing enabled?

File: indexer/services/socks/src/config.ts [hunk 2]
```diff
@@ -49,6 +49,10 @@ export const configSchema = {
   COMLINK_URL: parseString(),
   AXIOS_TIMEOUT_MS: parseInteger({ default: 5000 }), // 5 seconds
   INITIAL_GET_TIMEOUT_MS: parseInteger({ default: 20_000 }), // 20 seconds
+  BATCH_PROCESSING_ENABLED: parseBoolean({ default: false }),
```
Comment (Christopher-Li) [inline indexer/services/socks/src/config.ts:52]: nit: on by default?

**Review by Christopher-Li — APPROVED** [review]

File: indexer/services/socks/__tests__/lib/message-forwarder.test.ts (hunk 1 above)
Comment (dydxwill) [reply indexer/services/socks/__tests__/lib/message-forwarder.test.ts:185]: The ordering of functions in this test is not exactly like `index.ts`. `startConsumer` gets called in beforeAll, while the `messageForwarder` is initialized in each test. It's a bit of a pain to fix. I just set to false for now. If you feel strongly, I can add a TODO to fix this test.

File: indexer/services/socks/src/config.ts (hunk 2 above)
Comment (dydxwill) [reply indexer/services/socks/src/config.ts]: done

**Comment by dydxwill** [comment]
@mergifyio backport release/indexer/v5.x

**Pull request merged**